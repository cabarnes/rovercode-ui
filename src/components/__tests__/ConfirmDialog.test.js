import React from 'react';
import { shallow } from 'enzyme';
import { Button, Dialog } from '@material-ui/core';

import ConfirmDialog from '../ConfirmDialog';

describe('The ConfirmDialog component', () => {
  let onCancel;
  let onConfirm;
  let wrapper;

  beforeEach(() => {
    onCancel = jest.fn();
    onConfirm = jest.fn();

    wrapper = shallow(
      <ConfirmDialog
        title="Testing"
        open
        onConfirm={onConfirm}
        onCancel={onCancel}
        cancelButton="No"
        confirmButton="Yes"
      >
        Testing
      </ConfirmDialog>,
    );
  });

  test('renders on the page with no errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('calls cancel when cancelled', () => {
    wrapper.find(Button).at(0).simulate('click');

    expect(onCancel).toHaveBeenCalled();
    expect(onConfirm).not.toHaveBeenCalled();
  });

  test('calls cancel when exited', () => {
    wrapper.find(Dialog).simulate('close');

    expect(onCancel).toHaveBeenCalled();
    expect(onConfirm).not.toHaveBeenCalled();
  });

  test('calls confirm when confirmed', () => {
    wrapper.find(Button).at(1).simulate('click');

    expect(onCancel).not.toHaveBeenCalled();
    expect(onConfirm).toHaveBeenCalled();
  });
});
