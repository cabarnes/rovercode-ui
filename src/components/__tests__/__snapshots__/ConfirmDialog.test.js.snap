// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ConfirmDialog component renders on the page with no errors 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="confirm-dialog"
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="confirm-dialog"
  >
    Testing
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    Testing
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="default"
      onClick={[MockFunction]}
      variant="contained"
    >
      Yes
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[MockFunction]}
      variant="contained"
    >
      No
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
